GoogleForCreators/web-stories-wp
# Editor Redesign: Prepublish Panel

## Context

Redesign of pre publish panel, following figma: https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=221%3A13978 

## Summary

Updates pre publish panel to use design system. 

## Relevant Technical Choices

- Add a new button type to the design system that evaded my initial pass, we now have buttons that look like links. Kept this out of the normal button since we already have a link that looks like a button in there. 
- Per design specs, now you click on the "link" to go to the offending element. 
- I grabbed some changes in #6749 to the panel title area to visualize the updates to the panel title on the checklist, it's going to need more finesse. I'm marking the updates here that i did for that specifically. I am aware that the titles aren't quite right. 
- Pulled styles and static text out of `checklistTab` just to ease the readability of the file, it's big. 
- The designs use some padding that's not inline with the 8pt grid so I rounded to the nearest in most cases to keep that consistent. (The difference being things like 4px instead of 5px) 

## To-do

- Panel Titles 
- Scrollbar causing text shift in safari - got it fixed for chrome and firefox. Didn't want to sink more time into just safari since scrollbars have their own agenda and the panel titles still need to be updated. 
- Curious if we want to hang text with bullet points?
- Designs have issues grouped for pages, which I think is to say that pages 1, 2, 3 all need to have larger images, for example? However the functionality isn't set up for this so I ignored that and focused on just the style updates. 
- We might want to consider adding another section of text that's attached to the "link" buttons for screen readers to add some variety to what's being announced on those buttons other than "go to offending element".
- There were no updated designs for when there are no issues that need updating, so i just tweaked the existing ones to use the new theme. We might need a follow up ticket for that.

## User-facing changes

| Staging | This Branch | Designs or Empty |
| --- | --- | --- |
| ![Screen Shot 2021-03-18 at 2 23 33 PM](https://user-images.githubusercontent.com/10720454/111699582-8dc27200-87f5-11eb-98f0-1e6d37ef7330.png) | ![Screen Shot 2021-03-18 at 2 26 24 PM](https://user-images.githubusercontent.com/10720454/111699957-0295ac00-87f6-11eb-9274-98803dbb7f94.png) | <img width="326" alt="Screen Shot 2021-03-18 at 2 24 30 PM" src="https://user-images.githubusercontent.com/10720454/111699658-ab8fd700-87f5-11eb-87f4-aa472bc86eb2.png"> |
| ![Screen Shot 2021-03-18 at 2 25 06 PM](https://user-images.githubusercontent.com/10720454/111699741-c19d9780-87f5-11eb-90b7-52f4de86cbc0.png) | ![Screen Shot 2021-03-18 at 2 26 45 PM](https://user-images.githubusercontent.com/10720454/111699995-0d504100-87f6-11eb-8ee3-468adb96b210.png) | ![Screen Shot 2021-03-18 at 1 55 07 PM](https://user-images.githubusercontent.com/10720454/111702285-41793100-87f9-11eb-9aec-7246136ea947.png) |


| Right To Left | Left To Right | 
| --- | --- |
| https://user-images.githubusercontent.com/10720454/111700049-1e00b700-87f6-11eb-948a-fd8fc8d3903a.mp4 | https://user-images.githubusercontent.com/10720454/111700050-1e994d80-87f6-11eb-9c13-37aa534bf87e.mp4 |


## Testing Instructions

Look at the prepublish checklist and see that the designs match and functionality is preserved. 
"Links" are now the text to click to go to the offending list item. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6520
